src/commands/infra/generate.ts

```diff
@@ -244,22 +243,51 @@ export const validateRemoteSource = async (
   logger.info(
     `Checking if source repo: ${safeLoggingUrl} has been already cloned to: ${sourcePath}.`
   );
-
   try {
-    // Check if .git folder exists in ${sourcePath}, if not, then clone
-    // if already cloned, 'git pull'
-    if (fs.existsSync(path.join(sourcePath, ".git"))) {
-      await gitFetchPull(sourcePath, safeLoggingUrl);
```

Comment:
[change request] Can you please remove `gitFetchPull` function from this file and also its unit tests.

Reply:
By concatenating the git commands in a single function again (i.e. `repoClone`), will we reduce code coverage?

Reply:
Keeping code in Dennis's format with decoupled git functions